### DISCUSSION TITLE
Error when runing configure.sh

### DISCUSSION THREAD

**Discussion by Kazing** [organicmaps/organicmaps#2478] [Q&A]
Hello,
I am trying to build the app for android, and I followed all the steps in https://github.com/organicmaps/organicmaps/blob/master/docs/INSTALL.md

But I get an error when trying to run configure.sh (tried to run it with WSL and also with Visual studio Command prompt)
Therefore I can't build the app in android studio.
Here is details of the error I got : 
```
Initializing repository with default values in Open-Source mode.

Could not find a suitable toolset.

You can specify the toolset as the argument, i.e.:
    ./build.sh [options] gcc

Toolsets supported by this script are:
    acc, clang, como, gcc, intel-darwin, intel-linux, kcc, kylix, mipspro,
    pathscale, pgi, qcc, sun, sunpro, tru64cxx, vacpp

For any toolset you can override the path to the compiler with the '--cxx'
option. You can also use additional flags for the compiler with the
'--cxxflags' option.

A special toolset; cxx, is available which is used as a fallback when a more
specific toolset is not found and the cxx command is detected. The 'cxx'
toolset will use the '--cxx' and '--cxxflags' options, if present.

Options:
    --help                  Show this help message.
    --verbose               Show messages about what this script is doing.
    --debug                 Build b2 with debug information, and no
                            optimizations.
    --guess-toolset         Print the toolset we can detect for building.
    --cxx=CXX               The compiler exec to use instead of the detected
                            compiler exec.
    --cxxflags=CXXFLAGS     The compiler flags to use in addition to the
                            flags for the detected compiler.
```

**Comment by biodranik** [comment]
What is your OS and environment? The error message clearly says that you don't have a compiler (gcc, clang, MinGW, Visual Studio) in the environment.

**Reply by Kazing** [reply]
Windows 10, I have Android Studio & Visual Studio Code installed

**Reply by biodranik** [reply]
Please try to run it from Visual Studio Command Prompt (`bash configure.sh` if you have git for Windows installed), or run `./configure.sh` inside the WSL subsystem, as described in the documentation.

**Reply by Kazing** [reply]
Thanks for your answer,
Yes I have git installed, I tried again as you specified with Visual Studio Command Prompt & WSL command and I still have the same error as above.
Do you know what I should install more, since `configure.sh` doesn't find toolset?

**Reply by biodranik** [reply]
You need either a Visual Studio Community edition (or it's Build Tools) from here: https://visualstudio.microsoft.com/downloads/
Or you need to install clang++ or gcc++ in WSL. Did you try `apt get install clang++` or `apt get install g++`?
